server/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexResponse.java

```diff
@@ -50,30 +49,25 @@
     }
 
     protected static <T extends CreateIndexResponse> void declareFields(ConstructingObjectParser<T, Void> objectParser) {
-        declareAcknowledgedField(objectParser);
-        objectParser.declareField(constructorArg(), (parser, context) -> parser.booleanValue(), SHARDS_ACKNOWLEDGED,
-                ObjectParser.ValueType.BOOLEAN);
+        declareAcknowledgedAndShardsAcknowledgedFields(objectParser);
         objectParser.declareField(constructorArg(), (parser, context) -> parser.text(), INDEX, ObjectParser.ValueType.STRING);
```

Comment:
good one, I didn't see this, it should not happen as we never parse something that was serialized through the wire, and it can only be null before 5.6 where the high-level client didn't even exist yet. But the code allows for this so I am fixing and adding a test for it.